darts/timeseries.py

```diff
@@ -300,35 +837,100 @@ def gaps(self) -> pd.DataFrame:
         gap_df = pd.DataFrame()
         gap_df['gap_start'] = gap_starts
         gap_df['gap_end'] = gap_ends
+
+        def intvl(start, end):
+            if self._has_datetime_index:
+                return pd.date_range(start=start, end=end, freq=self._freq).size
+            else:
+                return start - end
+
         gap_df['gap_size'] = gap_df.apply(
-            lambda row: pd.date_range(start=row.gap_start, end=row.gap_end, freq=self.freq()).size, axis=1
+            lambda row: intvl(start=row.gap_start, end=row.gap_end), axis=1
         )
 
         return gap_df
 
-    def copy(self, deep: bool = True) -> 'TimeSeries':
+    def copy(self) -> 'TimeSeries':
         """
         Make a copy of this time series object
 
-        Parameters
-        ----------
-        deep
-            Make a deep copy. If False, the underlying pandas DataFrame will be the same
-
         Returns
         -------
         TimeSeries
             A copy of this time series.
         """
-        if deep:
-            return TimeSeries(self.pd_dataframe(), self.freq_str())
+        return TimeSeries(self._xa)  # the xarray will be copied in the TimeSeries constructor
+
+    def get_index_at_point(self, point: Union[pd.Timestamp, float, int], after=True) -> int:
+        """
+        Converts a point into an integer index
+
+        Parameters
+        ----------
+        point
+            This parameter supports 3 different data types: `pd.Timestamp`, `float` and `int`.
+
+            `pd.Timestamp` work only on series that are indexed with a `pd.DatetimeIndex`. In such cases, the returned
+            point will be the index of this timestamp if it is present in the series time index. It it's not present
+            in the time index, the index of the next timestamp is returned if `after=True` (if it exists in the series),
+            otherwise the index of the previous timestamp is returned (if it exists in the series).
+
+            In case of a `float`, the parameter will be treated as the proportion of the time series
+            that should lie before the point.
+
+            In the case of `int`, the parameter will returned as such, provided that it is in the series. Otherwise
+            it will raise a ValueError.
+        after
+            If the provided pandas Timestamp is not in the time series index, whether to return the index of the
+            next timestamp or the index of the previous one.
+
+        """
+        point_index = -1
```

Comment:
Necessary?

Reply:
No but it's only to avoid certain IDEs to complain that it might not be initialised...